### ISSUE TITLE
Trigger Name: Inconsistent about whether Spaces and special characters are allowed or not

### ISSUE THREAD

**Issue by dawwa** [microsoft/BotFramework-Composer#8240]
## Describe the bug
Spaces and special characters are not allowed when creating a Trigger, but are allowed to modify with Spaces and special characters.

![image](https://user-images.githubusercontent.com/44714043/124049063-90279100-d9cc-11eb-8439-9321bd1b3813.png)
![image](https://user-images.githubusercontent.com/44714043/124049174-c06f2f80-d9cc-11eb-8050-ccb964e7859f.png)

## Version
2.0

## OS

- [ ] macOS
- [X] Windows
- [ ] Ubuntu

**Comment by srinaath** [comment]
@dawwa Good catch. Essentially, the name of the trigger entered during trigger creation is actually the ID. The name that you edit from the form editor is the "display" name. It seems a bit confusing looking at the UI. @mewa1024 .

**Comment by dawwa** [comment]
I guess maybe we could consider enforcing a read only display name composed with the Id supplied by user for each item type, so users don't need to worry about Display name at all, even better, we could have dedicate icon for each item type in the Project Tree to make the clearer and more concise visual , similar to what the `Solution Explorer` looks like in `Visual Studio`.

**Comment by mewa1024** [comment]
What is the reason for prohibiting spaces in trigger names in the modal? Is that just an oversight, or was it a restriction that came from somewhere else? (Doesn't seem to be a LUIS restriction. I can add spaces to intent names in LUIS.ai)

(@JaskeeratBedi-cmk fyi)

**Comment by JaskeeratBedi-cmk** [comment]
@mareekuh : Tagging you for visibility
